### PR TITLE
Eliminate deployment metadata

### PR DESCRIPTION
**Pull request by tpdownes**

### CODE REVIEW

File: pkg/config/expand.go [hunk 1]
```diff
@@ -520,6 +520,15 @@ func (dc *DeploymentConfig) addDefaultValidators() error {
 	return nil
 }
 
+// FindAllIntergroupReferences finds all intergroup references within the group
+func (dg DeploymentGroup) FindAllIntergroupReferences(bp Blueprint) []Reference {
+	igcRefs := []Reference{}
```
Comment (mr0re1) [inline pkg/config/expand.go:525]: Let's make it a set.

File: pkg/shell/terraform.go [hunk 2]
```diff
@@ -181,19 +182,17 @@ func ExportOutputs(tf *tfexec.Terraform, metadataFile string, artifactsDir strin
...
 // working directory
-func ImportInputs(deploymentGroupDir string, metadataFile string, artifactsDir string) error {
-	deploymentRoot := path.Clean(path.Join(deploymentGroupDir, ".."))
-	thisGroup := path.Base(deploymentGroupDir)
+func ImportInputs(deploymentGroupDir string, artifactsDir string, expandedBlueprintFile string) error {
+	deploymentRoot := filepath.Clean(filepath.Join(deploymentGroupDir, ".."))
+	thisGroup := filepath.Base(deploymentGroupDir)
 
-	outputNamesByGroup, err := getIntergroupOutputNamesByGroup(thisGroup, metadataFile)
+	outputNamesByGroup, err := getIntergroupOutputNamesByGroup(thisGroup, expandedBlueprintFile)
```
Comment (mr0re1) [inline pkg/shell/terraform.go:189]: nit. `getIntergroupOutputNamesByGroup` and `GetDeploymentKinds` both export blueprint, it seems to be reasonable to export it once outside of those functions. Not for performance reason but, for reducing scope of functions.

File: tools/validate_configs/validate_golden_copy.sh [hunk 3]
```diff
@@ -70,6 +70,7 @@ run_test() {
 		rm -rf "${folder}/modules"
 	done
 	find . -name "README.md" -exec rm {} \;
+	sed -i -E 's/(ghpc_version: )(.*)/\1golden/' .ghpc/artifacts/expanded_blueprint.yaml
```
Comment (mr0re1) [inline tools/validate_configs/validate_golden_copy.sh:73]: nice.

File: pkg/shell/terraform.go (hunk 2 above)
Comment (tpdownes) [reply pkg/shell/terraform.go:189]: I had same thought. Would it better to have two funcs more like:

```
func (bp Blueprint) GetDeploymentKinds()

func (bp Blueprint) GetIntergroupOutputNamesByGroup(inputGroup string)
```

**Pull request closed without merging**